### DISCUSSION TITLE
LAG visibility between Parent and Child devices

### DISCUSSION THREAD

**Discussion by evlli** [netbox-community/netbox#6978] [General]
I've read this issues about this topic already, and I get the your take is that I should add all interfaces to the "parent" instead.
However, this is really /really/ unfeasable for us.
This would mean we have 500+ Interfaces on a single device. ( Aruba 6410 )
Not a big issue, this is workable.
But, we have many different configurations of 8 different Line Cards in different places in the chassis.
This is what kills me. I don't want to have tons of different templates for all combinations possible.

now that I have made my standpoint clear I have a question:
If we were to contribute this as a feature (the exact details of how it should be implemented are up for discussion) would it be accepted?

**Comment by candlerb** [comment]
> I've read this issues about this topic already, and I get the your take is that I should add all interfaces to the "parent" instead.
However, this is really /really/ unfeasable for us.

Then I'm afraid to say that perhaps Netbox is not the right tool for you.  The way that Netbox is designed does not permit "child devices" as a way of modelling line cards, and that is by design.

However, you don't need to template all possible combinations.  I suggest you have a single "Device Type" which is just the chassis.  When you create an instance of that Device, then you add the line cards as inventory items (if you want to track them), and at the same time you add interfaces directly to the Device.  The bulk interface-create function makes this easy: just enter `Ether1/[1-48]` exactly like that, and you'll get 48 identical interfaces created.  It's not ideal, but it *is* feasible.

If you want to make this easier for your users, you could write a small [Custom Script](https://netbox.readthedocs.io/en/stable/additional-features/custom-scripts/) which offers a set of choices and creates an appropriate device instance, with interfaces correctly populated.

> If we were to contribute this as a feature (the exact details of how it should be implemented are up for discussion) would it be accepted?

It depends what "feature" you're discussing.  Do you mean LAGs across child devices? I'm 100% sure that will be rejected. It has been made very clear in the past that child devices are not to be used to model line cards.

I think the long-term solution to this would be to have Inventory Item Types, and be able to attach device "components" (interfaces etc) to them.  That will be a major chunk of work though.

**Reply by evlli** [reply]
Okay, so if I understand this correctly, you are not opposed to the idea of modeling line cards (and sfp modules for that matter), you would just rather have it be a "Inventory Item Types" approach?

**Reply by candlerb** [reply]
Yes.  Not Devices, but something which models a module in a device.

**Reply by jeremystretch** [reply]
See #824, which sounds like what you're after.
